server/src/test/java/io/crate/integrationtests/PrivilegesIntegrationTest.java

```diff
@@ -733,4 +733,18 @@ public void testAccessesToPgConstraintEntriesWithRespectToPrivileges() throws Ex
         String superUserResult = printedTable(response.rows());
         assertThat(newUserWithPrivilegesResult, is(superUserResult));
     }
+
+    @Test
+    public void test_declare_fetch_close_for_non_super_users() {
+        try (Session testUserSession = testUserSession()) {
+            execute("BEGIN;", null, testUserSession);
+            execute("DECLARE my_cursor NO SCROLL CURSOR FOR SELECT * FROM pg_catalog.pg_attribute order by attname", null, testUserSession);
+            assertThat(response.rowCount(), is(0L));
+            execute("FETCH 1 FROM my_cursor;", null, testUserSession);
```

Comment:
`Fetch` and `Close` can only be tested once a cursor is declared, therefore it's tested in an integration test.

Reply:
It's possible to use `declare` in a unit test. See `CursorTest` for examples.